src/main/java/uk/gov/hmcts/reform/preapi/services/BookingService.java

```diff
@@ -114,4 +125,27 @@ public void markAsDeleted(UUID id) {
             bookingRepository.deleteById(id);
         }
     }
+
+    @Transactional
+    public UpsertResult shareBookingById(ShareBookingDTO shareBookingDTO) {
+        final var booking = bookingRepository.findById(shareBookingDTO.getBookingId())
+            .orElseThrow(() -> new NotFoundException("Booking: " + shareBookingDTO.getBookingId()));
+        final var sharedByUser = userRepository.findById(shareBookingDTO.getSharedByUserId())
+            .orElseThrow(() -> new NotFoundException("Shared by User: " + shareBookingDTO.getSharedByUserId()));
+        final var sharedWithUser = userRepository.findById(shareBookingDTO.getSharedWithUserId())
+            .orElseThrow(() -> new NotFoundException("Shared with User: " + shareBookingDTO.getSharedWithUserId()));
+
+        if (shareBookingRepository.existsById(shareBookingDTO.getId())) {
```

Comment:
Could move this to the top of the function

Reply:
Done